flow/probes/sflow.go

```diff
@@ -190,3 +183,136 @@ func NewSFlowProbe(a string, p int, g *graph.Graph) (*SFlowProbe, error) {
 
 	return probe, nil
 }
+
+type SkydivePacket struct {
+	gopacket.Packet
+	ci     gopacket.CaptureInfo
+	raw    *sflow.RawPacketFlow
+	sample *sflow.FlowSample
+}
+
+type SFlowReader struct {
+	gopacket.PacketSource
+	ProbeTarget      string
+	PacketChannel    chan SkydivePacket
+	RawPacketCounter uint64
+}
+
+func (r *SFlowReader) LinkType() layers.LinkType {
+	return layers.LinkTypeEthernet
+}
+
+func (r *SFlowReader) CreatePacket(sample *sflow.FlowSample, raw *sflow.RawPacketFlow) {
+	var ci gopacket.CaptureInfo
+	ci.Timestamp = time.Now()
+	ci.CaptureLength = int(raw.FrameLength)
+	ci.Length = int(raw.HeaderSize)
+
+	data := raw.Header
+	r.PacketChannel <- SkydivePacket{
+		gopacket.NewPacket(data, r.LinkType(), gopacket.DecodeOptions{}),
+		ci,
+		raw,
+		sample}
+}
+
+/* PacketSource */
+func (r *SFlowReader) NextPacket() (SkydivePacket, error) {
```

Comment:
It seems that this function is always returning nil as error so I think this should be removed